Polyfills/WebSocket/Source/WebSocket.cpp

```diff
@@ -0,0 +1,146 @@
+#include "WebSocket.h"
+#include <Babylon/JsRuntime.h>
+
+namespace Babylon::Polyfills::Internal
+{
+    void WebSocket::Initialize(Napi::Env env)
+    {
+        Napi::HandleScope scope{env};
+
+        static constexpr auto JS_WEB_SOCKET_CONSTRUCTOR_NAME = "WebSocket";
+
+        Napi::Function func = DefineClass(
+            env,
+            JS_WEB_SOCKET_CONSTRUCTOR_NAME,
+            {
+                StaticValue("CONNECTING", Napi::Value::From(env, 0)),
+                StaticValue("OPEN", Napi::Value::From(env, 1)),
+                StaticValue("CLOSING", Napi::Value::From(env, 2)),
+                StaticValue("CLOSED", Napi::Value::From(env, 3)),
+                InstanceAccessor("readyState", &WebSocket::GetReadyState, nullptr),
+                InstanceAccessor("url", &WebSocket::GetURL, nullptr),
+                InstanceAccessor("onopen", nullptr, &WebSocket::SetOnOpen),
+                InstanceAccessor("onclose", nullptr, &WebSocket::SetOnClose),
+                InstanceAccessor("onmessage", nullptr, &WebSocket::SetOnMessage),
+                InstanceAccessor("onerror", nullptr, &WebSocket::SetOnError),
```

Comment:
Setter with no getter is not typical. Why no getter?

Reply:
For setting up the event callbacks most websocket client implementations do not have an associated get method. For example, the default java websocket client: https://docs.oracle.com/javaee/7/api/index.html

Reply:
Getters have been added.